### PR TITLE
Fix StatusCode hard coded

### PR DESCRIPTION
**Pull request by orions**
Hi @suncat2000 ,

this PR fix the issue #107.

best,

### CODE REVIEW

File: Helper/DeviceView.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ class DeviceView
      * @param array        $configRedirect
      * @param RequestStack $requestStack
      */
-    public function __construct(array $configRedirect, RequestStack $requestStack = null)
+    public function __construct($configRedirect = null, RequestStack $requestStack = null)
```
Comment (orions) [inline Helper/DeviceView.php]: $configRedirect should not be default null.

File: Helper/DeviceView.php [hunk 2]
```diff
@@ -66,12 +63,20 @@ class DeviceView
...
+
+    /**
      * Constructor
      *
-     * @param \Symfony\Component\DependencyInjection\Container $serviceContainer
+     * @param array        $configRedirect
+     * @param RequestStack $requestStack
      */
-    public function __construct(RequestStack $requestStack = null)
+    public function __construct($configRedirect = null, RequestStack $requestStack = null)
```
Comment (xabbuh) [inline Helper/DeviceView.php]: Adding a new argument to the beginning of the parameters list is a BC break. To ease migration I would add it after existing argument.

File: Helper/DeviceView.php [hunk 3]
```diff
@@ -218,29 +223,28 @@ public function getSwitchParamValue()
...
     {
-        $statusCode = 302;
-
-        switch ($this->getSwitchParamValue()) {
+        switch($this->getSwitchParamValue()) {
             case self::VIEW_MOBILE:
-                return new RedirectResponseWithCookie($redirectUrl, $statusCode, $this->createCookie(self::VIEW_MOBILE));
             case self::VIEW_TABLET:
-                return new RedirectResponseWithCookie($redirectUrl, $statusCode, $this->createCookie(self::VIEW_TABLET));
+                return new RedirectResponseWithCookie($redirectUrl, $this->configRedirect[$this->getSwitchParamValue()]['status_code'], $this->createCookie($this->getSwitchParamValue()));
```
Comment (xabbuh) [inline Helper/DeviceView.php]: You have to check that the value is actually set.

File: Helper/DeviceView.php [hunk 4]
```diff
@@ -218,29 +223,28 @@ public function getSwitchParamValue()
...
             default:
-                return new RedirectResponseWithCookie($redirectUrl, $statusCode, $this->createCookie(self::VIEW_FULL));
+                return new RedirectResponseWithCookie($redirectUrl, $this->configRedirect[self::VIEW_FULL]['status_code'], $this->createCookie(self::VIEW_FULL));
```
Comment (xabbuh) [inline Helper/DeviceView.php]: same here

**Comment by xabbuh** [comment]
and maybe rename the argument to something like `$statusCodes`

**Comment by orions** [comment]
@xabbuh thank you for your feedback, next time i will not push a draft.

**Comment by xabbuh** [comment]
I don't think it's bad to push a draft as it allows you to get feedback early. :)

**Comment by orions** [comment]
@suncat2000 @xabbuh any other feedback ?

**Comment by orions** [comment]
hi, any news ?

File: Helper/DeviceView.php [hunk 5]
```diff
@@ -66,11 +63,17 @@ class DeviceView
...
+
+    /**
      * Constructor
      *
-     * @param \Symfony\Component\DependencyInjection\Container $serviceContainer
+     * @param RequestStack $requestStack
+     * @param array        $configRedirect
      */
-    public function __construct(RequestStack $requestStack = null)
+    public function __construct(RequestStack $requestStack = null, $configRedirect = null)
```
Comment (xabbuh) [inline Helper/DeviceView.php]: Now that you can inject config values into the constructor it IMO seems logical to be able to configure these values through the semantic config of the bundle.

Comment (orions) [inline Helper/DeviceView.php]: The bundle configuration is injected into the contructor, i don't understand, what do you expect, can you explain ?

File: Resources/config/mobile_detect.xml [hunk 6]
```diff
@@ -17,6 +17,7 @@
         <service id="mobile_detect.mobile_detector.default" class="%mobile_detect.mobile_detector.class%"/>
 
         <service id="mobile_detect.device_view" class="%mobile_detect.device_view.class%">
+            <argument>%mobile_detect.redirect%</argument>
```
Comment (suncat2000) [inline Resources/config/mobile_detect.xml]: @orions you need passed `%mobile_detect.redirect%` conf as second argument.
Your code
```
...
public function __construct(RequestStack $requestStack = null, $configRedirect = null)
...
```

File: Helper/DeviceView.php (hunk 5 above)
Comment (suncat2000) [inline Helper/DeviceView.php]: @orions and may be you move injection of `$configRedirect` from contructor  to call method? Similar as https://github.com/orions/MobileDetectBundle/blob/fd04f0c36395b86e0985f9c69f97d2be9ea1ccd5/Resources/config/mobile_detect.xml#L22
Something as:
```
<call method="setRedirectConfig">
    <argument>%mobile_detect. redirect%</argument>
</call>
```
That there was a similar config injection logic...

File: Resources/config/mobile_detect.xml (hunk 6 above)
Comment (xabbuh) [inline Resources/config/mobile_detect.xml]: you must swap the argument to match the constructor's signature

Comment (suncat2000) [inline Resources/config/mobile_detect.xml]: @xabbuh I asked @orions move injection of redirect config to a call method

Comment (orions) [inline Resources/config/mobile_detect.xml]: @suncat2000 i made the refactoring but travis is still running after more 1 hour.

Comment (orions) [inline Resources/config/mobile_detect.xml]: @suncat2000 ok problem solved with travis

**Comment by suncat2000** [comment]
@orions thanx for fixes

**Pull request merged**